sections_flask-jwt-extended/resources/user.py

```diff
@@ -0,0 +1,66 @@
+from flask_restful import Resource, reqparse
+from werkzeug.security import safe_str_cmp
+from flask_jwt_extended import create_access_token, create_refresh_token, jwt_refresh_token_required, get_jwt_identity
+from models.user import UserModel
+
+
+class UserRegister(Resource):
+    parser = reqparse.RequestParser()
+    parser.add_argument('username',
+                        type=str,
+                        required=True,
+                        help="This field cannot be blank."
+                        )
+    parser.add_argument('password',
+                        type=str,
+                        required=True,
+                        help="This field cannot be blank."
+                        )
+
+    def post(self):
+        data = self.parser.parse_args()
+
+        if UserModel.find_by_username(data['username']):
+            return {"message": "A user with that username already exists"}, 400
+
+        user = UserModel(data['username'], data['password'])
+        user.save_to_db()
+
+        return {"message": "User created successfully."}, 201
+
+
+class UserLogin(Resource):
+    parser = reqparse.RequestParser()
+    parser.add_argument('username',
+                        type=str,
+                        required=True,
+                        help="This field cannot be blank."
+                        )
+    parser.add_argument('password',
+                        type=str,
+                        required=True,
+                        help="This field cannot be blank."
+                        )
+
+    def post(self):
+        data = self.parser.parse_args()
+
+        user = UserModel.find_by_username(data['username'])
+
+        if user and safe_str_cmp(user.password, data['password']):
+            access_token = create_access_token(identity=user.id, fresh=True)
+            refresh_token = create_refresh_token(user.id)
+            return {
+                       'access_token': access_token,
+                       'refresh_token': refresh_token
+                   }, 200
+
+        return {"message": "Invalid Credentials!"}, 401
+
+
+class TokenRefresh(Resource):
+    @jwt_refresh_token_required
+    def post(self):
+        current_user = get_jwt_identity()
+        new_token = create_access_token(identity=current_user, fresh=False)
+        return {'access_token': new_token}, 200
```

Comment:
Are we still doing token refreshes?

Reply:
Yes, I removed fresh login, but we still have refresh token. I'll add some more notes to explain these terms.